proto/streamlit/proto/Block.proto

```diff
@@ -55,6 +56,15 @@ message Block {
     string icon = 3;
   }
 
+  message Dialog {
+    string title = 1;
+    bool dismissible = 2;
+    // we use string as width because the user is supposed to provide
+    //   predefined values (e.g. small, medium, ...)
+    optional string width = 3;
```

Comment:
the initial reason from the beginning of the project was to follow the same pattern as the DataEditor columns and type it on Python-side to nudge the user but still keep the possibility to force an arbitrary width if you really want / must to. Now that we settled on specific widths (`small` and `large`) I can type it more strongly on the proto-level; also👍 Also, for columns you might have more edge-cases where it could make sense to overrule the definition whereas for the dialog it should be pretty much standard

Reply:
I have updated the logic to have it as an enum in the proto now